src/Network/Freddy.hs

```diff
@@ -0,0 +1,48 @@
+{-# OPTIONS -XOverloadedStrings #-}
+module Network.Freddy (connect, respondTo) where
+
+import Network.AMQP
+import Data.Text (Text, pack)
+import Data.ByteString.Lazy.Char8 (ByteString)
+import Network.Freddy.Result
+
+connect :: String -> Text -> Text -> Text -> IO Connection
+connect host vhost user password =
+  openConnection host vhost user password
+
+respondTo :: Connection -> String -> (ByteString -> Either ByteString ByteString) -> IO ConsumerTag
```

Comment:
If the client calls
```haskell
respondTo connection "my-queue" myResponder
```
then `myResponder` can not do IO.
The responding mechanism cannot do IO, the type system disallows it.